### PR TITLE
[CoW Subsidy] Connect modal and add comp to swap mod

### PR DESCRIPTION
**Pull request by W3stside**
# Summary

Part of <TBD> and #2549 

Connects more data to modal and adds Fee discount comp to swap mod. Clickable to open

image

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @W3stside !

Nice changes! And here are my early issues: 
1. I think, it would be nice not to show this button to a not connected user 
image
image


2. Then, it would be great to display discount modal nicer in a mobile view (without blurred effect on the right side)
image

3. Maybe, we shouldn't show this fee discount when a user does not have (v)COW balance in the wallet? 
image

4. In a real mobile device, when I try opening the Fee discount tooltip, the modal is opened:
image

Thanks!

File: src/custom/components/CowBalance/index.tsx [hunk 1]
```diff
@@ -28,7 +28,7 @@ const CowBalance = ({ account, balance, title }: CowBalanceProps) => {
         </ClaimSummaryTitle>
       )}
 
-      {balance && !account && (
+      {balance && account && (
```
Comment (alfetopito) [inline src/custom/components/CowBalance/index.tsx:31]: 😌

File: src/custom/components/CowSubsidyModal/index.tsx [hunk 2]
```diff
@@ -43,15 +35,15 @@ export default function CowSubsidyModal({
   onDismiss,
   ...restProps
 }: Pick<ConfirmationModalProps, 'isOpen'> & Omit<ConfirmationModalContentProps, 'title' | 'topContent'>) {
-  const { chainId } = useActiveWeb3React()
+  const { account, chainId } = useActiveWeb3React()
 
-  const { activeClaimAccount } = useClaimState()
```
Comment (alfetopito) [inline src/custom/components/CowSubsidyModal/index.tsx:48]: 😌

**Review by alfetopito — APPROVED** [review]

**Comment by W3stside** [comment]
@elena-zh merging as consensus on slack for the showing was reached. Could you create an issue for @fairlighteth to style the other issues you found? thanks!

**Comment by elena-zh** [comment]
Hey @W3stside , I have created separate issues #2560 and #2561 for the UI cases I reported above. 
Returning back to the question about displaying modal to a not connected user/user with no balance, then I should say that at least we should show 0% discount, and 0 balance in the modal: 
image
image

Let me know please If I need to create a separate issue for this. 
Thanks!

**Comment by anxolin** [comment]
> https://github.com/orgs/gnosis/teams/gp-frontend thoughts? i think we should always show. 0 vcow? show. Not connected? Show

I would show always. 
For the balance, I think we can show 0 (and we could add a message saying, connect your wallet to check your discount). Alternativelly, replace the balance with the message.

**Comment by anxolin** [comment]
@W3stside this line doesn't make too much sense to me.  >0 u get 0?

image

**Comment by elena-zh** [comment]
@anxolin , it is fixed in #2558 PR :)
image

**Pull request merged**